RFP-01.md

```diff
@@ -0,0 +1,183 @@
+- Comprehensive documentation
+- Open-source code repository
+
+### Interface
+
+```solidity
+interface Auth {
+    /**
+     * Allows `usr` to perform root-level functions
+     */
+    function addOwner(address usr) external auth;
+    /**
+     * Denies `usr` from being able to perform root-level functions
+     */
+    function removeOwner(address usr) external auth;
+    /**
+     * Determine if `usr` is able to perform root-level functions
+     */
+    function isOwner(address usr) external view returns (bool);
+}
+
+interface SubPool is Auth {
+    /**
+     * Determine the amount required to `heal` the pool
+     * @return amt of immutable collateralToken
+     * @return cowAmt of COW
+     */
+    function dues() external view returns (uint256 amt, uint256 cowAmt);
+    /**
+     * Pull in from `msg.sender` the dues required to `heal` the pool
+     */
+    function heal() external;
+}
+
+interface SubPoolFactory is Auth {
+    // --- view functions ---
+
+    /**
+     * View function for determining the minimum collateral required if using `token`
+     * If the token is not a valid collateral, this function reverts.
+     * @param token The token to retrieve the collateral requirement for
+     */
+    function collateral(IERC20 token) external view returns (uint256);
+    /**
+     * Determine if an address has a collateralized sub-pool and has not announced an exit.
+     * While not enforced on-chain, an address for which this call returns `true` is a
+     * solver and is allowed to submit settlement on CoW Protocol through
+     * `SignedSettlement`.
+     * @param candidate The address to check
+     * @return bool if the address should be allowed to submit settlements through
+     * `SignedSettlement`
+     */
+    function canSolve(address candidate) external view returns (bool);
+    /**
+     * View function for determining the sub-pool for a given solver.
+     * Reverts if the input address has no sub-pool to its name.
+     * @param solver The solver to retrieve the sub-pool for
+     * @return The contract managing the collateral paid by this solver to be
+     * allowed to settle transactions on CoW Protocol
+     */
+    function solverSubPool(address solver) external view returns (SubPool);
+    /**
+     * View function for determining the backend URI for the sub-pool
+     * @return string URI
+     */
+    function backendUri(SubPool pool) external view returns (string memory);
+
+    // --- permissionless functions ---
+    /**
+     * Deploy a `SubPool` deterministically based on `msg.sender`.
+     * @notice The `msg.sender` SHOULD NOT be a contract.
+     * @param token The nominated token to use for collateral
+     * @param amt How much collateral to add to the sub-pool (the call will revert if
+     * this value is below the minimal collateral needed for `token`)
+     * @param cowAmt Amount of COW to add as collateral
+     * @param backendUri The URI to send the batches to the solver
```

Comment:
Do we want to have a pool address determined solely by the solver address? In case, I'd state it somewhere here.

Reply:
I've relaxed this a little to have the pool address determined with `CREATE` (and not deterministic) to allow for key rotations.